frontend/src/pages/modelRegistry/screens/ModelDetailsView.tsx

```diff
@@ -1,97 +1,103 @@
-            // eslint-disable-next-line no-console
-            console.log('TODO: save labels', editedLabels); // TODO API patch and refetch
-            return new Promise((resolve) => {
-              setTimeout(() => {
-                resolve();
-              }, 2000);
-            });
-          }}
-        />
-        <DashboardDescriptionListGroup
-          title="Properties"
-          action={
-            <Button isInline variant="link" icon={<PlusCircleIcon />} iconPosition="start">
-              Add property
-            </Button>
-          }
-          isEmpty // TODO
-          contentWhenEmpty="No properties"
-        >
-          TODO properties here
-        </DashboardDescriptionListGroup>
-      </DescriptionList>
-    </FlexItem>
-    <FlexItem flex={{ default: 'flex_1' }}>
-      <DescriptionList isFillColumns>
-        <DashboardDescriptionListGroup
-          title="Model ID"
-          isEmpty={!rm.externalID}
-          contentWhenEmpty="No model ID"
-        >
-          <ClipboardCopy hoverTip="Copy" clickTip="Copied" variant="inline-compact">
-            {rm.externalID}
-          </ClipboardCopy>
-        </DashboardDescriptionListGroup>
-        <DashboardDescriptionListGroup title="Owner">
-          <RegisteredModelOwner registeredModelId={rm.id} />
-        </DashboardDescriptionListGroup>
-        <DashboardDescriptionListGroup
-          title="Last modified at"
-          isEmpty={!rm.lastUpdateTimeSinceEpoch}
-          contentWhenEmpty="Unknown"
-        >
-          <ModelTimestamp timeSinceEpoch={rm.lastUpdateTimeSinceEpoch} />
-        </DashboardDescriptionListGroup>
-        <DashboardDescriptionListGroup
-          title="Created at"
-          isEmpty={!rm.createTimeSinceEpoch}
-          contentWhenEmpty="Unknown"
-        >
-          <ModelTimestamp timeSinceEpoch={rm.createTimeSinceEpoch} />
-        </DashboardDescriptionListGroup>
-      </DescriptionList>
-    </FlexItem>
-  </Flex>
-);
+const ModelDetailsView: React.FC<ModelDetailsViewProps> = ({ registeredModel: rm }) => {
+  const { apiState, refreshAPIState } = React.useContext(ModelRegistryContext);
+  return (
+    <Flex
+      direction={{ default: 'column', md: 'row' }}
+      columnGap={{ default: 'columnGap4xl' }}
+      rowGap={{ default: 'rowGapLg' }}
+    >
+      <FlexItem flex={{ default: 'flex_1' }}>
+        <DescriptionList isFillColumns>
+          <EditableTextDescriptionListGroup
+            title="Description"
+            contentWhenEmpty="No description"
+            value={rm.description || ''}
+            saveEditedValue={(value) =>
+              apiState.api
+                .patchRegisteredModel(
+                  {},
+                  // TODO remove the getPatchBody* functions when https://issues.redhat.com/browse/RHOAIENG-6652 is resolved
+                  getPatchBodyForRegisteredModel(rm, { description: value }),
+                  rm.id,
+                )
+                .then(refreshAPIState)
```

Comment:
Why do you want to call refresh api here? This just reload the fetching of the CR, which is not needed when PATCH right?

We should refresh ONLY the version details here, we are reloading pretty much everything.

Reply:
For this you can get the `refresh` value form `useModelVersionById` that we get in `ModelVersionDetails` and pass it to the component.

Reply:
I've just created a commit (cause i wanted to test it) which is more self-explanatory, I'll leave it here so you can check the suggestion: https://github.com/lucferbux/odh-dashboard/commit/6b921040d92a6d6cb0b366e8ac81a1fcb65051d1

Reply:
Ah, good call thank you! I was looking at `apiState` for a way to do that, of course it makes sense that it would be on `useModelVersionById`